packages/api/src/hooks/useRemoteConfig.ts

```diff
@@ -15,12 +16,25 @@ const remoteConfigQuery = async function () {
 };
 
 function useRemoteConfig(enabled = false) {
-    return useQuery({
-        queryKey: ['remoteConfig'],
-        queryFn: remoteConfigQuery,
-        initialData: initData,
-        enabled,
-    });
+    const [data, setData] = useState(initData);
+
+    useEffect(() => {
+        enabled &&
+            remoteConfigQuery()
+                .then(async res => {
+                    const resHash = await ObjectUtils.hashObject(res);
+                    const dataHash = await ObjectUtils.hashObject(data);
+                    if (resHash !== dataHash) {
+                        setData(res);
+                    }
+                })
+                .catch(error => {
+                    // eslint-disable-next-line no-console
+                    console.log('Remote Config error: ', error);
+                });
+    }, [enabled]);
```

Comment:
do we need data in the dependency array for the comparison with hashObject? as for comparison, we need the latest data from the state. If the enabled flag never changes after the hook call, useEffect will never get the latest data available in the state rather it'll always have the data which useEffect got at the first time for comparison, and it may be the initData only

Reply:
@shahzaib-deriv Don't think so, if I add data to the dependency array then after line #28 this `useEffect` will be called 2nd time, because data changed, but we don't need it. About `initData`, yes, if `enabled` is false then we will have `initData` as a return result, and it's expected if `enabled` is false